envs/quadrotors/wrappers/reward_shaping.py

```diff
@@ -49,7 +48,7 @@ def step(self, action):
             env_reward_shaping[key] = weight
 
         obs, rewards, dones, infos = self.env.step(action)
-        if self.is_multiagent:
+        if hasattr(self.env, 'is_multiagent') and self.env.is_multiagent is True:
```

Comment:
As I said, it is better to cache the result of this whole expression (e.g. save it to the class variable), since hasattr queries are rather slow.
Also, instead of `self.env.is_multiagent is True` just write `self.env.is_multiagent`

I will approve for now but please change this in the following commit.